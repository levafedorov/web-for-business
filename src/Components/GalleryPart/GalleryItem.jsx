import React from 'react';



export default function GalleryItem(props){

   const {photo, title, width, active} = props;

    
   const bgImageStyle = {
       width: `${width}%`,
       backgroundImage: `url("${photo}")`
   }

   const titleUp = {
       top:0
   }

   return <div className="gallery-item" style={bgImageStyle}>
       <div className="title" style={active ? titleUp : null}>
           <h3>{title}</h3>
       </div>
   </div>
}


GalleryItem.defaultProps = {
     photo: '',
     title: 'No photo',
     width: "0",
     active: true
}
