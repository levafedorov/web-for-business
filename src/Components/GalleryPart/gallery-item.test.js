import React from 'react';
import {shallow} from 'enzyme';
import GalleryItem from './GalleryItem';


const fakeProps = {photo: "fake url", title: "fake title",  width:"100", active:true}

describe('GalleryItem component', () => {
    
    describe('Check GalleryItem props', ()=>{
        test("Check with props",  () => {
            const component = shallow(<GalleryItem {...fakeProps}/>)
            expect(component).toMatchSnapshot();
        });
        test("Check with defaultProps",  () => {
            const component = shallow(<GalleryItem/>)
            expect(component).toMatchSnapshot();
        });
    });

});