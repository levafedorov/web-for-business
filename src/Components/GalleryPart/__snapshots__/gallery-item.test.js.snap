// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryItem component Check GalleryItem props Check with defaultProps 1`] = `
<div
  className="gallery-item"
  style={
    Object {
      "backgroundImage": "url(\\"\\")",
      "width": "0%",
    }
  }
>
  <div
    className="title"
    style={
      Object {
        "top": 0,
      }
    }
  >
    <h3>
      No photo
    </h3>
  </div>
</div>
`;

exports[`GalleryItem component Check GalleryItem props Check with props 1`] = `
<div
  className="gallery-item"
  style={
    Object {
      "backgroundImage": "url(\\"fake url\\")",
      "width": "100%",
    }
  }
>
  <div
    className="title"
    style={
      Object {
        "top": 0,
      }
    }
  >
    <h3>
      fake title
    </h3>
  </div>
</div>
`;
